File: backend/src/main/java/com/example/backend/service/PostService.java

package com.example.backend.service;

import com.example.backend.dto.PostDto;

import java.util.List;

public interface PostService {

    void createPost(PostDto postDto);
    void deletePost(Long id);

    PostDto getPost(Long id);
    List<PostDto> getPostsOfFollowedUsers(Long userId);
    List<PostDto> getPostsByUser(Long userId);
    void likePost(Long postId, Long userId);
    void unlikePost(Long postId, Long userId);
    int getLikes(Long postId);

}

File: frontend/src/app/services/post.service.ts
// src/app/core/services/post.service.ts

import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Post } from '../models/post.model';
import { environment } from '../../environments/environment';

@Injectable({
  providedIn: 'root'
})
export class PostService {
  private apiUrl = `${environment.apiUrl}/api/posts`;

  constructor(private http: HttpClient) {}

  getPostsByUser(userId: number): Observable<Post[]> {
    return this.http.get<Post[]>(`${this.apiUrl}/user/${userId}`);
  }

  getPost(postId: number): Observable<Post> {
    return this.http.get<Post>(`${this.apiUrl}/${postId}`);
  }

  createPost(post: Post): Observable<Post> {
    return this.http.post<Post>(this.apiUrl, post);
  }

  deletePost(postId: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${postId}`);
  }

  likePost(postId: number, userId: number): Observable<void> {
    return this.http.post<void>(`${this.apiUrl}/${postId}/like`, { userId });
  }

  unlikePost(postId: number, userId: number): Observable<void> {
    return this.http.post<void>(`${this.apiUrl}/${postId}/unlike`, { userId });
  }
}
